### DISCUSSION TITLE
"mcsim" not responding on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by mudwalkercj** [Moddable-OpenSource/moddable#1097] [Getting started] [answered]
I have followed your Moddable SDK – Getting Started instructions and all goes well until I get to Step 10
    cd $MODDABLE/examples/helloworld
    mcconfig -d -m -p lin

The helloworld example shows in RH panel of xsbug, but the mcsim window comes up with a red block in the RH panel and then a pop up tells me that mcsim is not responding. Fans on the laptop speed up and stay that way until I quit mcsim.

I then tried to update as per instructions with the same result.

Any ideas what might be causing this and how to fix it?

TIA Colin J

**Comment by phoddie** [comment]
Thanks for the report. Linux can be a little challenging because there are so many different configurations. We'll take a look and get back to you.

**Comment by mkellner** [comment]
Hello Colin.
Can you please send a screen shot? I am interested in both the xsbug and mcsim windows.

**Comment by mudwalkercj** [comment]
Here you go...
I cannot move the Error pop-up in the mcsim window, only quit
![2023-04-27_19-38](https://user-images.githubusercontent.com/125122937/234961411-e9fc062f-d3f9-4d18-a738-68f8c12b99ad.png)

**Reply by mkellner** [reply]
Thank you. Push the "Play: button in xsbug.

**Reply by mudwalkercj** [reply]
Nor sure what is supposed to happen but...
![2023-04-27_19-49](https://user-images.githubusercontent.com/125122937/234962809-24a3aa6e-7fd8-44b8-a9fa-72370987b63b.png)

**Reply by mkellner** [reply, marked as answer]
The simulator is paused as the debugger has it stopped at the `debugger;` point in helloworld. When you pressed "Play", the debugger allows the app to continue, mcsim continues it's startup process and the "Hello World - sample" appears in the console.
That is all expected and shows it is working.

For a more interesting example, instead of helloworld, try the balls app:

`cd $MODDABLE/examples/piu/balls`
`mcconfig -d -m -p lin`

**Reply by mudwalkercj** [reply]
Excellent!  Might be worth adding that into the Instructions.  

Appreciate your very prompt response and look forward to using Moddable alongside Node-RED and some ESP and RPi devices.!

Thank you,
Colin

**Reply by mkellner** [reply]
I agree. I will add some better guidance in the instructions.
